src/choleskyQR.jl

```diff
@@ -0,0 +1,141 @@
+function cholesky_jacobimatrix(w::Function, P::OrthogonalPolynomial)
+    !(P isa Normalized) && error("Polynomials must be orthonormal.")
+    W = Symmetric(P \ (w.(axes(P,1)) .* P)) # Compute weight multiplication via Clenshaw
+    bands = CholeskyJacobiBands(W, P)
+    return SymTridiagonal(bands[1,:],bands[2,:])
+end
+function cholesky_jacobimatrix(W::AbstractMatrix, P::OrthogonalPolynomial)
+    !(P isa Normalized) && error("Polynomials must be orthonormal.")
+    !(W isa Symmetric) && error("Weight modification matrix must be symmetric.")
+    bands = CholeskyJacobiBands(W, P)
+    return SymTridiagonal(bands[1,:],bands[2,:])
+end
+
+# The generated Jacobi operators are symmetric tridiagonal, so we store their data as two bands
+mutable struct CholeskyJacobiBands{T} <: AbstractCachedMatrix{T}
```

Comment:
Why is there both a `data` and an `array`? Explain what the fields are.

Reply:
see the other comment on this